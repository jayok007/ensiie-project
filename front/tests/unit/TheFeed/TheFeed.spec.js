import { shallowMount } from '@vue/test-utils'
import TheFeed from '@/views/TheFeed'

describe('TheFeed', () => {
  it('should render properly when user have keywords', async () => {
    const wrapper = shallowMount(TheFeed, {
      mocks: {
        $store: {
          state: {
            user: {
              id: 1,
              pseudo: 'Test',
              keywords: ['a', 'b', 'c']
            }
          },
          getters: {
            isAuthenticated: true
          },
          commit: jest.fn()
        },
        $api: {
          getUserInfo () {
            return Promise.resolve()
          }
        }
      },
      data () {
        return {
          bars: [
            {
              id: 1,
              name: 'Bar 1',
              address: '1 rue',
              keywords: ['b'],
              photoreference: 'ref',
              rating: '4.4'
            }
          ]
        }
      }
    })
    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
