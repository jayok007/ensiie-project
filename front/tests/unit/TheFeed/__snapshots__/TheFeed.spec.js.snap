// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheFeed should render properly when user have keywords 1`] = `
<vcontainer-stub tag="div" fluid="" grid-list-xl="">
  <vlayout-stub tag="div" row="" wrap="">
    <vflex-stub tag="div" sm12="" md8="" xl6="">
      <vcard-stub tag="div" class="elevation-4">
        <vcardtitle-stub class="headline">
          Bienvenue sur votre FeedBar Test
        </vcardtitle-stub>
        <vcardtext-stub>
          <h4>Vos mots clés sont</h4>
          <vchip-stub outline="true" value="true" color="primary">

          </vchip-stub>
          <vchip-stub outline="true" value="true" color="primary">

          </vchip-stub>
          <vchip-stub outline="true" value="true" color="primary">

          </vchip-stub>
        </vcardtext-stub>
        <vcardactions-stub>
          <vbtn-stub activeclass="v-btn--active" flat="true" tag="button" type="button" to="/me" color="secondary">Modifier mes mots clés</vbtn-stub>
          <vbtn-stub activeclass="v-btn--active" flat="true" tag="button" type="button" to="/" color="secondary">Recherche personnalisée</vbtn-stub>
        </vcardactions-stub>
      </vcard-stub>
    </vflex-stub>
    <bar-stub id="1" name="Bar 1" address="1 rue" photoreference="ref" rating="4.4"></bar-stub>
  </vlayout-stub>
</vcontainer-stub>
`;
